src/Lykke.Common.Validation/ImageType/ImageTypeAttribute.cs

```diff
@@ -0,0 +1,113 @@
+        {
+            if (imageExtensions == null || imageExtensions.Length == 0)
+            {
+                _imageTypes = DefaultImageTypes;
+            }
+            else
+            {
+                foreach (var extension in imageExtensions)
+                {
+                    if (DefaultImageTypes.TryGetValue(extension, out var imageType))
+                    {
+                        _imageTypes.Add(extension, imageType);
+                    }
+                }
+            }
+        }
+
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+            var file = (IFormFile) value;
+
+            if (file == null)
+                return new ValidationResult("Value is null or empty.");
+
+
+            var extension = GetExtension(file.FileName);
+
+            if (!_imageTypes.TryGetValue(extension, out var imageType))
+                return new ValidationResult("File extension is invalid.");
+
+            using (var fileStream = file.OpenReadStream())
+            {
+                var longestSignature = imageType.HexSignatures.OrderByDescending(x => x.Length).FirstOrDefault();
+
+                if (longestSignature == null)
+                    return new ValidationResult("No signatures specified for type.");
+
+                var buffer = new byte [longestSignature.Length];
+                fileStream.Read(buffer, 0, buffer.Length);
+
+                if (!ValidateSignature(buffer, imageType.HexSignatures))
+                    return new ValidationResult("File is not an image, signature is invalid.");
+
+                return ValidationResult.Success;
+            }
+        }
+
+        private static bool ValidateSignature(byte[] file, IEnumerable<string> validHexSignatures)
+        {
+            if (file == null || validHexSignatures == null)
+                return false;
+
+            return validHexSignatures.Select(StringToByteArray)
+                .Any(byteSequence => file.Take(byteSequence.Length).SequenceEqual(byteSequence));
+        }
+
+        private static byte[] StringToByteArray(string hex)
+        {
+            var numberChars = hex.Length;
+            var bytes = new byte[numberChars / 2];
+            for (var i = 0; i < numberChars; i += 2)
+                bytes[i / 2] = Convert.ToByte(hex.Substring(i, 2), 16);
+            return bytes;
+        }
+
+        private static string GetExtension(string fileName)
+        {
+            return string.IsNullOrWhiteSpace(fileName) ? string.Empty : Path.GetExtension(fileName);
+        }
+
+        private static IDictionary<string, ImageType> InitDefaultImageTypes()
+        {
+            var defaultImageTypesDict = new Dictionary<string, ImageType>();
+
+            foreach (var imageType in DefaultImageTypesList)
+            foreach (var extension in imageType.Extensions)
+                defaultImageTypesDict.Add(extension, imageType);
```

Comment:
Is it safe to add without checking? For example two different types have the same extension

Reply:
Types are define in Validation library, two different types can't have same extension.